### ISSUE TITLE
Opt-out of Auto-update Downloader

### ISSUE THREAD

**Issue by OiYouYeahYou** [desktop/desktop#3410]
Currently, you are using an auto-update that downloads in the background with no option to prevent the auto-download. This becomes a problem in two cases:

1. **Memory and Processor hog:** The auto-updater package hogs a lot of system resources, this becomes a concern on systems with lower specs. In my case, this actually renders my development laptop unusable while the updater is working.

1. **Limited Data Plans:** More importantly, for people on a limited data plan, routinely downloading a ~70MB electron package can eat up the data allowance. This has happened to me, unfortunately.

My request is to add an opt-out setting to prevent the auto-updater being called. Instead, requesting and comparing the version information. Then notifying the user a new version is out and requesting they download it. This enables the end-user to prevent data allowances and system resources hogging.

**Comment by shiftkey** [comment]
@OiYouYeahYou thanks for the feedback!

> The auto-updater package hogs a lot of system resources, this becomes a concern on systems with lower specs. In my case, this actually renders my development laptop unusable while the updater is working.

I'd love to try this out myself and see how bad it feels. Could you tell me about your setup?

**Comment by OiYouYeahYou** [comment]
Excessive Disk I/O is something I neglected to mention.

Specs are:
- 4 GB RAM
- Dual 2.1 GHz
- ~< 5 MB/s Duplex Hard drive

Software overhead:
- VS Code
- Firefox
- _Sometimes Chrome_
- Discord
- Github Desktop

Whenever the electron updater runs (other software included) to even check for an update it causes a noticeable slowdown. I've also recognised this on friends systems where the worst offender is Discord, even if it's just checking. A quick taskkill of the update process will usually fix this issue though.

**Comment by shiftkey** [comment]
@OiYouYeahYou that's some great information - thanks for sharing! For reference we're using the inbuilt Electron updater APIs - based on [Squirrel.Windows](https://github.com/Squirrel/Squirrel.Windows) - but I'm sure there's room for improvement (especially on CPU and I/O usage by the sounds of it).

**Comment by HackcatDev** [comment]
It'll be also nice to download only updated files, not the complete package. Most likely main GUI won't be updated in new version and Electron framework is also static. I've seen this updating model in TLauncher that updates Minecraft to newest version using only 5-6 megabytes of data when complete package requires about 80 MB to download

**Comment by shiftkey** [comment]
> It'll be also nice to download only updated files, not the complete package.

@HackcatDev the framework we use in Electron on Windows, [`Squirrel.Windows`](https://github.com/Squirrel/Squirrel.Windows/), supports delta packages to apply just what has changed. Unfortunately there's no equivalent feature in `Squirrel.Mac` currently.

**Comment by mapninja** [comment]
+1 on this issue. It seems like I probably have type my extremely complex and secure admin password for Github updates 3 or 4 times a day, lately, and more than that since I get it wrong 1 or 2 times each go around. It would be great to have the option to opt out of this.

**Comment by Catfriend1** [comment]
+1 for turning off remote updates. OpenSource projects should make trust by allowing the user to decide.

**Comment by Croydon** [comment]
I'm disagreeing strongly with the new feature #750. It should be my own decision when deleting branches.

Since there doesn't seem to be any way to disable auto-updates this is forcing me to abandon GitHub Desktop completely 😞 

**Edit:** On Windows deleting `Update.exe` actually does the trick

**Edit 2 (2026-02-02):** It is probably best to just rename the `Update.exe` file instead of deleting it, for the case that updates should be re-enabled at some point in the future.

**Comment by jakubpolomsky** [comment]
Please turn of the updater or put the git binaries into a separate directory so that I don't have to modify my path every time

**Comment by Peter92** [comment]
Each time an update rolls out, everyone is our company who uses Desktop is blocked until IT manually whitelists the new executables for each person, since they're using path based firewall rules.
To avoid having to urgently ask them to drop what they're doing and fix things, we've had to resort to renaming `Update.exe` for now.

**Comment by Neustradamus** [comment]
Very old ticket.

Any news on it?

How to disable auto-update?
- Desktop
- Atom

**Comment by zearp** [comment]
After 3 years, why isn't there a toggle for this yet? I don't want apps to download updates without asking me. Specially when I'm on a metered connection. An app that (without asking) downloads full new versions and then on the next start even upgrades itself is rather bad. What if I don't want to update to the new version for whatever reason? Why isn't there an option to disable auto-update checks or at the very least disable auto-downloading and installing these updates?

**Comment by i300220** [comment]
My version installed in dec 2020 was running perfectly fine on windows 7 64 until it autoupdated to 2.6.2 today.  It never restarted until I reboot.  When ran, it did not exit properly and required another reboot.  I finally uninstalled it, reinstalled it  to retest (bad), and finally got rid of it completely.  Wasted 3 hrs on this.

See https://github.com/desktop/desktop/issues/11455

**Comment by salwa445** [comment]
3410

**Comment by pragmatta** [comment]
I experience a performance regression going from 2.6.4 -> 2.6.6 (https://github.com/desktop/desktop/issues/9193) and would like to disable updates until it's fixed.

**Comment by i300220** [comment]
> I experience a performance regression going from 2.6.4 -> 2.6.6 (#9193) and would like to disable updates until it's fixed.

Here's what I did to stop auto updates.  Hope it helps.
https://github.com/desktop/desktop/issues/11455#issuecomment-770424076

**Comment by wireddown** [comment]
An app that automatically updates itself also undermines [`choco` managed installations](https://community.chocolatey.org/packages/github-desktop).

**Comment by amark** [comment]
How do I disable auto-update of GitHub Desktop app itself?

**Comment by i300220** [comment]
> How do I disable auto-update of GitHub Desktop app itself?

https://github.com/desktop/desktop/issues/11455#issuecomment-770424076

**Comment by zearp** [comment]
It’s clear GitHub doesn’t care, look at the age of this issue. We’re not allowed to disable auto updates, if you’re in metered connections too bad. If you don’t like it stop using GitHub. Which I might just do. Move my tiny repos to Gitlab and call it a day.

**Comment by Niko-O** [comment]
**Bump**

**Comment by kamauwanjoroge** [comment]
This needs to be merged. Losing up to 126 MB of my data plan per update.

**Comment by michaelhart** [comment]
Removing this app from my workflow because gross mandatory automatic updates does not work for me.

**Comment by erikhatfield** [comment]
> Removing this app from my workflow because gross mandatory automatic updates does not work for me.

Agreed. 

Will look into selectively blocking some network calls.

**Comment by grandeljay** [comment]
Another issue is that I am using a package manager to update my software (chocolatey). So I don't need GitHub Desktop to autoupdate and mess with my ecosystem

**Comment by zwbetz-gh** [comment]
Following this. 

As others mentioned, my MacBook fan kicks up quite hard when the auto-updater runs. 

It's not an issue when I'm actively using my computer. But, when I walk into the office and unexpectedly hear the fan running -- that's an issue.

**Comment by touw22** [comment]
+1 on this.
I have a separate administrator account I use to manually authorize any software updates (a common security stance for most companies).  Every day I have multiple github desktop requests I have to approve and it's quite annoying.

**Comment by mackonsti** [comment]
There is indeed no way to avoid auto-updating, this is not convenient and rather frustrating.
The only way found is to disable exectutable rights from `Electron Framework` via MacOS Terminal....

**Comment by Niko-O** [comment]
Jesus christ, this issue is over 4 years old and people still keep coming here because it's still relevant. How is this still not fixed?
Why do we still have to use the wonky workaround @Croydon mentioned above?

**Comment by zwbetz-gh** [comment]
> Why do we still have to use the wonky workaround @Croydon mentioned above?

Peeked, and that one only works for Windows. 

I’d be satisfied if there was a workaround for Mac too. (If someone is reading this and you already know one for Mac, please do share).

**Comment by zwbetz-gh** [comment]
I don't want to sound ungrateful -- I really like this app as a visual git differ. 

But I do have to poke fun at this :)

https://github.com/desktop/desktop/blob/881e8c5fb49dad5943c5de95ad0bb9f838840179/app/src/ui/terms-and-conditions/terms-and-conditions.tsx#L120-L127

Specifically:

> If you choose to use the Service

**Comment by isaacatmann** [comment]
+1 on the importance of this, we work in secured environments where users aren't given read/write access to /Applications or to install update apps on their own.  All updates come from IT, this annoyingly prompts them daily.  Most apps have a config setting that can be pushed to disable built in updates.

**Comment by isaacatmann** [comment]
> There is indeed no way to avoid auto-updating, this is not convenient and rather frustrating.
> The only way found is to disable exectutable rights from `Electron Framework` via MacOS Terminal....

@mackonsti Did this cause any other side effects to the application?  

Assuming you mean this file: /Applications/GitHub Desktop.app/Contents/Frameworks/Electron Framework.framework/Versions/Current

**Comment by mackonsti** [comment]
Hi @isaacatmann for my simpler, hobbyist needs, no there were no side effects as far as I can see (on Catalina).

Yes I mean this same file (prefer the direct link rather than the symbolic one)
`/Applications/GitHub Desktop.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework`

**Comment by egandrew** [comment]
Not a good scenario for users who don't have perms to install new software.  Have to remove app.  I have dozens of pen testing apps and open source.  This is the only app the intrusively pop up a window, including waking it up from sleep, to light up my office.  Really sad to hear a development group of such a great product just dismiss the need.  I'm sure I'll be back, but will have to stick this app in a win VM as it's a disturbance in a mac environment where you "don't have admin creds to install a helper without requesting temp perms".

**Comment by probablykasper** [comment]
Another problem with auto-updates is when it forces you into a new version that has [a bug that causes constant crashes (and that is not even going to be fixed because it's not important enough)](https://github.com/desktop/desktop/issues/14075). Auto-updates basically help make sure that old versions become unusable.

It's totally fine if my OS/environment isn't supported anymore, but I would at least appreciate if you don't forcibly brick the app

**Comment by Johnjohnnyjingles** [comment]
Why àm í seeing thís text

**Comment by ahurasid** [comment]
> > It'll be also nice to download only updated files, not the complete package.
> 
> @HackcatDev the framework we use in Electron on Windows, [`Squirrel.Windows`](https://github.com/Squirrel/Squirrel.Windows/), supports delta packages to apply just what has changed. Unfortunately there's no equivalent feature in `Squirrel.Mac` currently.

 **@#142
Duplicate of #91** # # #

- [ ]

**Comment by the-real-tokai** [comment]
I may want to add that the auto-updating also tries to install a newer version that may be incompatible with an older version of the OS. Basically the **application will entirely destroy itself** for no real reason whatsoever (the old version still works fine for the required jobs at hand).

Almost happened to me here on a Mac OS X 10.10 "Yosemite" installation. Just by pure luck I prevented it when I cancelled the suspicious request to install some "helper tool". But if inspect the cache location for the downloaded update, the new application bundle is incompatible with Yosemite (stop symbol on the icon).

**Comment by EricFromCanada** [comment]
@the-real-tokai I've found I can prevent it from breaking itself by restoring the [last working version](https://desktop.githubusercontent.com/github-desktop/releases/2.9.15-55730bca/GitHubDesktop-x64.zip) and enabling its "Locked" flag in Finder.

**Comment by EricFromCanada** [comment]
An alternative approach: exploit the app's dislike of [Gatekeeper Path Randomization](https://mjtsai.com/blog/2016/06/16/gatekeeper-path-randomization/). (Only applies to macOS Sierra and later, though.)
1. Download the [last working version](https://desktop.githubusercontent.com/github-desktop/releases/2.9.15-55730bca/GitHubDesktop-x64.zip) and move the zip archive to `/Applications`
2. use Archive Utility to extract the zip archive in-place
3. verify that the app still has the quarantine flag applied:

        $ xattr /Applications/GitHub\ Desktop.app
        com.apple.quarantine
4. Open the app, select _GitHub Desktop > About GitHub Desktop_, and click "Check for Updates". The app should complain bitterly with a message saying "Cannot update while running on a read-only volume". Such a shame; I'll get right on that.

**Comment by splorp** [comment]
Per @EricFromCanada’s suggestion (which works like a darn, by the way).

> Download the last working version …

For me, the last working version under High Sierra was [3.0.0](https://desktop.githubusercontent.com/github-desktop/releases/3.0.0-b9edf65/GitHubDesktop-x64.zip)

**Comment by the-real-tokai** [comment]
@ErikFromCanada

The "Locked" flag is more or less what I did too, albeit I locked down its update cache directory instead and protected it from accidentally messing it up (when cleaning the Cache directory, f.ex.)

```
cd ~/Library/Caches/
mkdir -p com.github.GitHubClient.ShipIt
sudo chown root:wheel com.github.GitHubClient.ShipIt
sudo chmod -rwx com.github.GitHubClient.ShipIt
sudo chflags uchg com.github.GitHubClient.ShipIt
```

Blocking access to "raw.githubusercontent.com" (via an outgoing firewall like LittleSnitch) seems to work too, but seems to break user avatars too in the process (if that's something you care about).

The latest version that seems to work properly still on Yosemite w/o throwing strange errors seems to be 2.8.0:
https://desktop.githubusercontent.com/github-desktop/releases/2.8.0-853674e1/GitHubDesktop-x64.zip

Hope the information helps someone. I wish it wasn't that awkwardly time consuming. Just an option to disable auto-updates, protection against self-updating to incompatible versions, and easy access to older versions via a download archive… why is that so hard? :)

**Comment by rbn30117** [comment]
It seems the maintainers checked out of this topic. I am also on a limited data plan and having automatic updates is detrimental to my data plan.

**Comment by pablco** [comment]
+1

**Comment by steveward** [comment]
From https://github.com/desktop/desktop/issues/16022:

>Beginning in version 3.1.5, it appears that a setting to automatically check for updates on program launch was added with no way to disable the functionality. I do my work in an offline environment, so this results in a nuisance prompt every time I launch the program. Please add an option to disable automatic update checking.

**Comment by Niko-O** [comment]
> Thanks @DaJoNel, I'll add this to #3410 where we are ~~tracking~~ ignoring this.

Excuse me for being passive-aggressive about this, but it's been over 4 years!

**Comment by DaJoNel** [comment]
Oddly enough, I never experienced this before updating to 3.1.5, and I’ve been using this for years. I see it’s been a problem for others a lot longer.

**Comment by djw-do** [comment]
Any movement on adding a flag/option for this?

**Comment by jessenic** [comment]
We use GitHub Desktop app also in a offline environment. How to disable the "Update check failed" popup? It's annoying and every time it happens, it layers on top of the previous one. So after a few days of having GitHub Desktop open I have to click the "Dismiss" button a dozen times to make the popup disappear.

**Comment by zearp** [comment]
I don't want to be a Debby Downer but I don't think Github cares about this issue at all.

This issue has been open for 6 years and Github with all their money and employees was unable to add two simple toggles for checking updates and automatically installing them.

Github will not fix this as they simply do not care. I've stopped using Github Desktop and I'm thinking about migrating to Codeberg or some other place where they actually listen to their users.

**Comment by zearp** [comment]
> Disabling auto-update is not in our roadmap, sorry 😕

https://github.com/desktop/desktop/issues/16227#issuecomment-1449596959

Might as well close this issue.

**Comment by RVNJ** [comment]
> Per @EricFromCanada’s suggestion (which works like a darn, by the way).
> 
> > Download the last working version …
> 
> For me, the last working version under High Sierra was [3.0.0](https://desktop.githubusercontent.com/github-desktop/releases/3.0.0-b9edf65/GitHubDesktop-x64.zip)

For me, I'm running OS El Capital (10.11.6) - just recently updated and got locked out of the program and had to download an older version _from an off-site_ might I add.

Version 3.0.6 seems to work for this OS but I'll check later to see if I can get anything more up-to-date than that running.

For now though, pressing ⌘-I on the application will open an info panel where you can tick the 'Locked' checkbox. This will prevent the app from auto-updating but won't stop it from trying to download (sadly). When it fails to update it will give you an error or two but you can ignore them.

**Comment by fredericschmidt** [comment]
Problem on OS 10.12.6 running Github Desktop >3.2.0

This is the easiest way, by far, thanks a lot !

> For now though, pressing ⌘-I on the application will open an info panel where you can tick the 'Locked' checkbox. This will prevent the app from auto-updating but won't stop it from trying to download (sadly). When it fails to update it will give you an error or two but you can ignore them.

**Comment by jimrs** [comment]
jesus christ, six years

**Comment by Only4Gamer** [comment]
> > Per @EricFromCanada’s suggestion (which works like a darn, by the way).
> > > Download the last working version …
> > 
> > 
> > For me, the last working version under High Sierra was [3.0.0](https://desktop.githubusercontent.com/github-desktop/releases/3.0.0-b9edf65/GitHubDesktop-x64.zip)
> 
> For me, I'm running OS El Capital (10.11.6) - just recently updated and got locked out of the program and had to download an older version _from an off-site_ might I add.
> 
> Version 3.0.6 seems to work for this OS but I'll check later to see if I can get anything more up-to-date than that running.
> 
> For now though, pressing ⌘-I on the application will open an info panel where you can tick the 'Locked' checkbox. This will prevent the app from auto-updating but won't stop it from trying to download (sadly). When it fails to update it will give you an error or two but you can ignore them.

In latest version pressing command+I open issue in browser. Do you know the keyboard shortcut for the latest version? Or maybe you now know a better way to turn off auto-update?

**Comment by yereverluvinunclebert** [comment]
+1 for turning off remote updates. MUST be optional. If you need to know reasons then you haven't thought it through.

For the moment I renamed update.exe to arseUpdate.exe

**Comment by EricFromCanada** [comment]
> In latest version pressing command+I open issue in browser

In Finder, have the GitHub Desktop application selected, and either press ⌘-I or select _File > Get Info_.

**Comment by fredjohnsen** [comment]
> I may want to add that the auto-updating also tries to install a newer version that may be incompatible with an older version of the OS. Basically the **application will entirely destroy itself** for no real reason whatsoever (the old version still works fine for the required jobs at hand).
> 
> Almost happened to me here on a Mac OS X 10.10 "Yosemite" installation. Just by pure luck I prevented it when I cancelled the suspicious request to install some "helper tool". But if inspect the cache location for the downloaded update, the new application bundle is incompatible with Yosemite (stop symbol on the icon).

Wait... you mean the newest version of some software you use isn't perfectly compatible with a 9 YEAR OLD VERSION OF YOUR OPERATING SYSTEM?!?!?!?!?!? What is this world coming to?

**Comment by probablykasper** [comment]
@fredjohnsen No, the problem is that old versions of the software don't work with old versions of the OS. You're being very disrespectful, please stop and apologize

**Comment by yereverluvinunclebert** [comment]
Running with an older version of an operating system is a perfectly reasonable choice for any number of reasons. The comment from fred just serves to obfuscate. The issue is whether there should be an opt out for auto-updates. Of course there should be. If anyone needs an explanation I ask them to think for a while, the reasons ought to come automatically with a working brain.

There is a workaround to the auto-update for the moment, renaming the update.exe, so the priority is not quite so high on this one.

**Comment by jimrs** [comment]
> > I may want to add that the auto-updating also tries to install a newer version that may be incompatible with an older version of the OS. Basically the **application will entirely destroy itself** for no real reason whatsoever (the old version still works fine for the required jobs at hand).
> > Almost happened to me here on a Mac OS X 10.10 "Yosemite" installation. Just by pure luck I prevented it when I cancelled the suspicious request to install some "helper tool". But if inspect the cache location for the downloaded update, the new application bundle is incompatible with Yosemite (stop symbol on the icon).
> 
> Wait... you mean the newest version of some software you use isn't perfectly compatible with a 9 YEAR OLD VERSION OF YOUR OPERATING SYSTEM?!?!?!?!?!? What is this world coming to?
> 
> What kind of people expect these things to work?

i-is that a whopping 9 years old software??? what has the world come to, people using such outdated and obsolete software??? everyone knows that software older than 1 year cannot be functional... AND REMEMBER THE SECURITY!!!

**Comment by benguild** [comment]
does anyone know the hostname to block to prevent these updates from happening?

**Comment by Niko-O** [comment]
@benguild 
Sorry, I don't know the answer, but just in case you didn't see it in this sea of comments: @Croydon found a workaround for Windows:
> **Edit:** On Windows deleting `Update.exe` actually does the trick

If you are willing to invest a bit of time, you might be able to use Telerik Fiddler to find the relevant domains upon program start.

**Comment by TurtleWilly** [comment]
> does anyone know the hostname to block to prevent these updates from happening?

On the Mac (e.g. via Little Snitch) you block connection it does via its "Github Desktop Helper" component (the update is loaded over `raw.githubusercontent.com`). The main application then still can access repositories via `github.com` itself (for fetch, etc.)

**Comment by benguild** [comment]
> > does anyone know the hostname to block to prevent these updates from happening?
> 
> On the Mac (e.g. via Little Snitch) you block connection it does via its "Github Desktop Helper" component (the update is loaded over `raw.githubusercontent.com`). The main application then still can access repositories via `github.com` itself (for fetch, etc.)

That domain's not specific to that, though. I was hoping to figure out if there was a way to block the updates themselves from even being considered, but not sure.

**Comment by TurtleWilly** [comment]
> That domain's not specific to that, though. I was hoping to figure out if there was a way to block the updates themselves from even being considered, but not sure.

That's the only thing it seems to load from that domain though. Blocking it works fine and the unwanted and broken updates stopped. The rest of the functions related to repositories continue to work fine.

If you wanted to block the domain globally via `/etc/hosts` then that wouldn't work, of course.

**Comment by hiiamboris** [comment]
After reading this thread, I got rid of GH Desktop out of solidarity :)

**Comment by cr-jkb** [comment]
I got here because the latest GitHub Desktop Version has issues with the merge dialog and some other functions. I waited in hope for a fix but the next fixing update takes already multiple weeks to arrive.

Therefore I searched for an Auto-Update Disable on Github Desktop because some of my machines still have older GitHub Desktop Versions installed, which however update themselves automatically when I use them. But once I can open and use the older version on each machine. On Restart the update is already installing, which I'd like to be able to bypass in the future

**Comment by bakomon** [comment]
+1

**Comment by alainbryden** [comment]
I work in a corporate environment where once GitHub Desktop auto-updates itself, the new executable is blocked by group policy because it wasn't installed via the standard process. We need an option to defer the update so that it can be installed via the correct process.

**Comment by thquinn** [comment]
![image](https://github.com/desktop/desktop/assets/1022438/eb3b2822-3175-4a79-8a46-910f3ef8fe8e)

I find myself looking at this dialog multiple times a week: does Github for Windows need so many updates? Why do they take so long to apply?

**Comment by AkashLiquiLoans** [comment]
I'm really unsure about what it's installing. I receive updates every other day, and it's quite frustrating.

**Comment by AkashDPA** [comment]
I'm really unsure about what it's installing. I receive updates every other day, and it's quite frustrating.

**Comment by SmartManoj** [comment]
> **Edit:** On Windows deleting `Update.exe` actually does the trick

**Workaround:**
```python
import os
file = r'%LOCALAPPDATA%\GitHubDesktop\Update.exe'
file = os.path.expandvars(file)
print('Renaming file:', file)
# auto-updating github desktop without user consent is annoying when there is a bug 
funny_name = file.replace('Update.exe', 'AnnoyingUpdate.exe')
os.rename(file, funny_name)
print('Successfully renamed file')
```

**Comment by alitahir6001** [comment]
This is still an issue in 2024. I hadn't fired up GH Desktop in literal ages and immediately my machine was just frozen because of all the system resources the auto update started to hog. Super annoying. A 6 minute task turns into 3x the wait time just so i can see my mouse move again. I'm uninstalling, this is dumb.

**Comment by darvi-sh** [comment]
I open the app, do some work, it starts eating through my mobile data. I close the app, it warns me that the automatic update is ongoing, but I don't have time or the data to spare. Same cycle repeats and my data is chewed through.

The uncontrollable automatic updates of this (cr)app **does not honor Low Data mode**.

What a terrible software engineering by Microsoft.

**Comment by samtupy** [comment]
+1000 on disabling auto updates.

I'm a totally blind computer user who recently started to enjoy Github Desktop because of it's ease regarding browsing repository history and the beautiful interface the program has for committing to repositories.

If I was browsing through a list of github commits, I could tell how close to the top of the list I was at because my screen reader would announce things like "1 of 100", "2 of 100", "3 of 100" etc after describing the commit I arrowed over in the history window.

And then I launched Github desktop today to discover to my horror that the accessible counter that let me see where I was in the list of commits was just... gone. Previous version would deselect the currently focused commit or something so I could instantly tell if new commits appeared after pull because my screen reader would say "not selected", that stopped happening too.

Delete update.exe you say?

too late bro, accessibility already broken.

This one's minor, but what about the future? What happens when I'm in a rush to commit and then launch the app and oops random accessibility problem for my morning!

Just wanted to add one more reason to the already massive list of reasons in this thread to disable automatic updates.

Might stop using this software, have been very unimpressed with what I see in this thread. So many good reasons for disabling auto update yet "just delete update.exe after the software is already broken for you haha hopefully you have an older installer around" is the only response from the developers I seem to find here.

I don't mind beta testing software and reporting accessibility bugs but there is a difference between that and watching my workflow randomly break because I don't get to choose when the software I use might casually become less accessible!

**Comment by literary-programmer** [comment]
Seriously, this recent bug was hilarious. This thing is broken with both jaws and nvda.

**Comment by jaknz** [comment]
> After 3 years, why isn't there a toggle for this yet? I don't want apps to download updates without asking me. Specially when I'm on a metered connection. An app that (without asking) downloads full new versions and then on the next start even upgrades itself is rather bad. What if I don't want to update to the new version for whatever reason? Why isn't there an option to disable auto-update checks or at the very least disable auto-downloading and installing these updates?

Another four years later, totalling seven so far, and we see how much Microsoft cares for the community they bought. Especially around accessibility re: the comment a couple above mine, they obviously just DGAF about anyone who isn't in their expected use case.

**Comment by BigMacAdmin** [comment]
Please, allow us to manage the update experience for our users. Prompting for an administrator password (macOS) 7 times in less than 60 seconds is absurd.

**Comment by twilightgod** [comment]
Still not fixed? Unbeliable, it keeps poping up from my MacOS, looks like the developer team themselves never use their own software

**Comment by AkashDPA** [comment]
@shiftkey  Hey! I'm planning to add a flag under Settings > Advanced labeled "Disable Auto Update" Could you confirm if this approach works for you?

Alternatively, can provide an option to edit the value of UpdateCheckInterval, which is currently set to 4 hours.

**Comment by tidy-dev** [comment]
@AkashDPA Thanks for wanting to contribute. But this is not PR that the Desktop team will merge at this time.

**Comment by Niko-O** [comment]
@tidy-dev Boooooo!

![Image](https://github.com/user-attachments/assets/b034f43f-72e8-4e50-bff9-dfdeba65c379)

**Comment by RVNJ** [comment]
> @AkashDPA Thanks for wanting to contribute. But this is not PR that the Desktop team will merge at this time. 

Worst comment in this thread by far, as evidenced by everyone's responses. Refusing to fix something that locks users out of your application for no reason is disgusting behavior.

**Comment by RVNJ** [comment]
> I don't want to be a Debby Downer but I don't think Github cares about this issue at all.
> 
> 
> 
> This issue has been open for 6 years and Github with all their money and employees was unable to add two simple toggles for checking updates and automatically installing them.
> 
> 
> 
> Github will not fix this as they simply do not care. I've stopped using Github Desktop and I'm thinking about migrating to Codeberg or some other place where they actually listen to their users.

Literally the worst example of wontfix I've ever seen. Doubly so considering the very obvious community dissent.

**Comment by AkashDPA** [comment]
@tidy-dev @niik  Based on the activity on this issue, it seems like it needs to be addressed.

Could you provide some guidance on how the feature should be implemented? For example, should there be a disable toggle, with a forced auto-check on every app startup?

I'm happy to develop the solution—feel free to merge it whenever the timing works best for you.

**Comment by SmartManoj** [comment]
https://github.com/SmartManoj/GitHub-Desktop-Annoying-Update-Fixer

**Comment by RVNJ** [comment]
> [@tidy-dev](https://github.com/tidy-dev) [@niik](https://github.com/niik) Based on the activity on this issue, it seems like it needs to be addressed.
> 
> Could you provide some guidance on how the feature should be implemented? For example, should there be a disable toggle, with a forced auto-check on every app startup?
> 
> I'm happy to develop the solution—feel free to merge it whenever the timing works best for you.

They don't want this fixed.* They will not approve any solution presented to them and absolutely will not merge with main because it does not do what they want it to; that is, to force people to stay 'up-to-date' for a resource that has zero reason to enforce such a thing.

*As evidenced by this quote:
* > [@AkashDPA](https://github.com/AkashDPA) Thanks for wanting to contribute. But this is not PR that the Desktop team will merge at this time.

**Comment by RVNJ** [comment]
> https://github.com/--

**Always** be wary of these sorts of links. Do not trust **any** code that you have not **already read though**. I'm not saying that this is an example of malicious code, but be sure to do your due diligence.

**Comment by Croydon** [comment]
@SmartManoj What problem does this solve? Downloading and running a Python script to rename a single file takes longer than just renaming said file.

**Comment by SmartManoj** [comment]
First version https://github.com/desktop/desktop/issues/3410#issuecomment-2334232639

Running a script is faster than going to that folder and renaming.

---

After certain days, if you want to update (recently they added filter files & generate commit msg with copilot), this GUI will come in handy to enable and disable updates. (You can integrate it with other apps like AuoHotKey)

**Comment by ChromaPIE** [comment]
LMAO THE 3.5 UPDATE IS A WHOLE ASS REGRESSION AND THEY STILL WANNA KEEP THEIR AUTO-UPDATE FORCED ON

cmon github let us celebrate the decade anniversary of this issue sure you will.

**Comment by agustinov** [comment]
AutoUpdate BREAKS the current installed version and after closing current version new version doesn't start (even when trying to start manually). Just give users option to turn-off this f**bugging autoupdate.

**Comment by DoomyDwyer** [comment]
Fed up

**Comment by DoomyDwyer** [comment]
Such a simple thing and you can't be bothered to do it.

**Comment by huntfx** [comment]
If you won't ever disable auto update, at least consider updating the error message so it's not so doom and gloom.

> Error: The application is missing a dependency it needs to check and install updates. This is very, very bad.

It's not "very, very bad", it's intentional for a lot us as we've had to manually rename Update.exe.

**Comment by xalt7x** [comment]
> **Edit:** On Windows deleting `Update.exe` actually does the trick

That also breaks uninstaller.  
So I assume that the best way is to make GitHub Desktop "portable"

1. Install some required version (without Internet connection, just to make sure)
2. Archive  %LocalAppData%\GitHubDesktop directory excluding Update.exe
3. Uninstall GitHub Desktop
4. Unpack an archive and create/pin shortcut to GitHubDesktop.exe somewhere

**Comment by bakomon** [comment]
> So I assume that the best way is to make GitHub Desktop "portable"

You can use this portable version: [#3707 (comment)](https://github.com/desktop/desktop/issues/3707#issuecomment-1803014584)

**Comment by xalt7x** [comment]
> > So I assume that the best way is to make GitHub Desktop "portable"
> 
> You can use this portable version: [#3707 (comment)](https://github.com/desktop/desktop/issues/3707#issuecomment-1803014584)

@bakomon
Thanks! [ghdPortable](https://portableapps.com/node/68346) seem decent.
Guess, I will switch to their launcher/wrapper, if official "GitHub Desktop" becomes more dependent on registry.
But for now the script to automate steps I've mentioned above, seem sufficient.
```
@echo off

SET GHD_install_PATH=%LocalAppData%\GitHubDesktop
SET GHD_Desktop_Shortcut=%UserProfile%\Desktop\GitHub Desktop.lnk
SET GHD_StartMenu_Shortcut=%AppData%\Microsoft\Windows\Start Menu\Programs\GitHub, Inc\GitHub Desktop.lnk

echo:
echo Saving GitHub Desktop directory + Shortcuts
robocopy "%GHD_install_PATH%" "%GHD_install_PATH%_copy" /mir /nfl /ndl /njs /njh
IF EXIST "%GHD_Desktop_Shortcut%" ( copy "%GHD_Desktop_Shortcut%" "%GHD_Desktop_Shortcut%_copy")
IF EXIST "%GHD_StartMenu_Shortcut%" ( copy "%GHD_StartMenu_Shortcut%" "%GHD_StartMenu_Shortcut%_copy")
echo:
echo Uninstalling GitHub Desktop and removing its directory
"%GHD_install_PATH%\Update.exe" --uninstall -s
rmdir /S /Q "%GHD_install_PATH%
echo:
echo Restoring GitHub Desktop directory + Shortcuts
echo Also removing Update.exe to avoid auto-updates
move "%GHD_install_PATH%_copy" "%GHD_install_PATH%"
move "%GHD_Desktop_Shortcut%_copy" "%GHD_Desktop_Shortcut%"
move "%GHD_StartMenu_Shortcut%_copy" "%GHD_StartMenu_Shortcut%"
DEL %GHD_install_PATH%\Update.exe
echo:
pause
```

**Comment by friday** [comment]
adding `127.0.0.1 central.github.com` to `/etc/hosts` seems to work

(but if you need/use central.github.com to download the app, or if your package manager does, then it's a problem).

**Comment by AnJoMort0** [comment]
I would also appreciate if auto-update could be opted-out for a totally different reason.
I have a small, very old computer that I use to bring around to do small works in public transport etc and the newest versions are not supported in the MacOs version in that computer. I need to reinstall an older version every time I open the app.

**Comment by TouhouJevil** [comment]
How is this not implemented already AFTER 3 YEARS, disabling auto updates is one of the most basic features any program should have now days, do i need to remind how much people complain about being unable to disable for example windows auto updates? not everyone likes being forced on their face to update, just add a setting that let people choose between manually updating or auto updating this shouldn't be rocket science. on the middle of 2026.

**Comment by hiiamboris** [comment]
> choose between manually updating or auto updating this shouldn't be rocket science. on the middle of 2026.

Auto updates = control = money. If they didn't shove Copilot down your throats, you wouldn't use it.

**Comment by RVNJ** [comment]
> I would also appreciate if auto-update could be opted-out for a totally different reason. I have a small, very old computer that I use to bring around to do small works in public transport etc and the newest versions are not supported in the MacOs version in that computer. I need to reinstall an older version every time I open the app.

I have a solution [here](https://github.com/desktop/desktop/issues/3410#issuecomment-1506133385) if you'd like to try it. Let me know if it works, but that's what I'm using on my 'old' MacBook Pro.

**Comment by gk-pew** [comment]
Being able to disable auto-update is essential when deploying GitHub Desktop to user endpoints via Intune/Jamf/etc. My MDM manages the updates, the end users don't have local admin rights, I'm tired of them asking for local admin to update when the GitHub Desktop app checks in the background/prompts the user for credentials they don't have every time an update comes out and NO way to disable that check/prompt and eliminate the unnecessary support tickets/help desk calls/etc.

Being able to disable auto update is a BASIC functionality for a vast majority of applications.... GitHub needs to get with the program, its been _years_

**Comment by RVNJ** [comment]
> Being able to disable auto-update is essential when deploying GitHub Desktop to user endpoints via Intune/Jamf/etc. My MDM manages the updates, the end users don't have local admin rights, I'm tired of them asking for local admin to update when the GitHub Desktop app checks in the background/prompts the user for credentials they don't have every time an update comes out and NO way to disable that check/prompt and eliminate the unnecessary support tickets/help desk calls/etc.
> 
> Being able to disable auto update is a BASIC functionality for a vast majority of applications.... GitHub needs to get with the program, its been _years_

It's also just shitty practice, not that they give a flying fuck.
